modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -199,212 +199,161 @@ public TableManager(
-                        ((ExtendedTableConfiguration) tablesCfg.tables().get(tblName)).schemas()
-                                .listenElements(new ConfigurationNamedListListener<>() {
-                                    @Override
-                                    public CompletableFuture<?> onCreate(ConfigurationNotificationEvent<SchemaView> schemasCtx) {
-                                        long causalityToken = schemasCtx.storageRevision();
-
-                                        if (!busyLock.enterBusy()) {
-                                            fireEvent(
-                                                    TableEvent.ALTER,
-                                                    new TableEventParameters(causalityToken, tblId, tblName),
-                                                    new NodeStoppingException()
-                                            );
-
-                                            return CompletableFuture.failedFuture(new NodeStoppingException());
-                                        }
+                UUID tblId = tblCfg.id().value();
 
-                                        try {
-                                            // Avoid calling listener immediately after the listener completes to create the current table.
-                                            // FIXME: https://issues.apache.org/jira/browse/IGNITE-16369
-                                            if (ctx.storageRevision() != schemasCtx.storageRevision()) {
-                                                return tablesByIdVv.get(causalityToken).thenAccept(tablesById -> {
-                                                    TableImpl table = tablesById.get(tblId);
+                String tblName = tblCfg.name().value();
 
-                                                    ((SchemaRegistryImpl) table.schemaView())
-                                                            .onSchemaRegistered(
-                                                                    SchemaSerializerImpl.INSTANCE.deserialize(
-                                                                            (schemasCtx.newValue().schema())));
+                SchemaDescriptor schemaDescriptor = SchemaSerializerImpl.INSTANCE.deserialize((schemasCtx.newValue().schema()));
 
-                                                    fireEvent(TableEvent.ALTER, new TableEventParameters(causalityToken,
-                                                            table), null);
+                if (!busyLock.enterBusy()) {
+                    if (schemaDescriptor.version() != INITIAL_SCHEMA_VERSION) {
+                        fireEvent(
+                                TableEvent.ALTER,
+                                new TableEventParameters(causalityToken, tblId, tblName),
+                                new NodeStoppingException()
+                        );
```

Comment:
It is a fail event, this can be thrown just after an exception has happened.
But if you are asking about an ordinal result when schema registered. It is a place where we may through earlier than the VersinedValue completed, because the corresponding object (a table) is already available and get the schema immediately.